src/entities/RfqmJobEntity.ts

```diff
@@ -10,11 +10,42 @@ export interface StoredFee {
 }
 
 export enum RfqmJobStatus {
-    InQueue = 'inQueue',
-    Processing = 'processing',
-    Submitted = 'submitted',
-    Successful = 'successful',
-    Failed = 'failed',
+    // Transaction has been enqueued and will be processed once a worker is available
+    PendingEnqueued = 'pending_enqueued',
+    // Transaction has passed initial validation. Last look will be executed and transaction will be submitted if last look is accepted.
+    PendingProcessing = 'pending_processing',
+    // Transaction has passed initial verification and has been submitted to the mem pool
+    PendingSubmitted = 'pending_submitted',
+    // Transaction has been resubmitted to the network with an updated gas price
+    PendingResubmittedGas = 'pending_resubmitted_gas',
+    // Transaction has been resubmitted to the network for an uncategorized reason
+    PendingResubmittedUncategorized = 'pending_resubmitted_uncategorized',
+
+    // Transaction has expired prior to eth call or worker is not available to make an eth call
+    FailedExpired = 'failed_expired',
+    // Transaction does not contain call data
+    FailedValidationNoCallData = 'failed_validation_no_call_data',
+    // Transaction does not include a maker URI
+    FailedValidationNoMakerUri = 'failed_validation_no_maker_uri',
+    // Transaction does not contain an order
+    FailedValidationNoOrder = 'failed_validation_no_order',
+    // Transaction does not contain a fee
+    FailedValidationNoFee = 'failed_validation_no_fee',
+    // Eth Call made before transaction submission was unsuccessful
+    FailedEthCallFailed = 'failed_eth_call_failed',
+    // Market Maker declined the last look
+    FailedLastLookDeclined = 'failed_last_look_declined',
+    // Submitting the transaction to the network was unsuccessful
+    FailedSubmitFailed = 'failed_submit_failed',
+    // Transaction was reverted
+    FailedReverted = 'failed_reverted',
+    // Transaction failed for an uncategorized reason
+    FailedUncategorized = 'failed_uncategorized',
```

Comment:
Small thought - over time, if we add statuses, the ordering of this may lose its meaning. Thoughts on just making it alphabetical? 

```suggestion
    // Eth Call made before transaction submission was unsuccessful
    FailedEthCallFailed = 'failed_eth_call_failed', 
    // Transaction has expired prior to eth call or worker is not available to make an eth call
    FailedExpired = 'failed_expired', 
    // Market Maker declined the last look
    FailedLastLookDeclined = 'failed_last_look_declined', 
    // Transaction was reverted
    FailedReverted = 'failed_reverted', 
    // Submitting the transaction to the network was unsuccessful
    FailedSubmitFailed = 'failed_submit_failed', 
    // Transaction failed for an uncategorized reason
    FailedUncategorized = 'failed_uncategorized', 
    // Transaction does not contain call data
    FailedValidationNoCallData = 'failed_validation_no_call_data', 
    // Transaction does not contain a fee
    FailedValidationNoFee = 'failed_validation_no_fee', 
    // Transaction does not include a maker URI
    FailedValidationNoMakerUri = 'failed_validation_no_maker_uri', 
    // Transaction does not contain an order
    FailedValidationNoOrder = 'failed_validation_no_order', 
```

Personal preference, but I do feel that the statuses are fairly self descriptive and do not need comments. Additionally, the comments may lie if they are later used in different points in the lifecycle. (So my slight pref would be to remove them)

Reply:
10000% onboard with alphabetizing. I actually read this manifesto this person wrote about how alphabetizing is the only correct ordering, even when you have a field like 'id' or something you really want to put at the top. I got radicalized and now I can't find it.. been looking forever.

10000% will fight you about the comments. I think for the most part I agree they are self explanatory,  but also what's self explanatory for me or you may not be for the next person to touch the code. I think your point about the comments lying is even more telling. If a usage of the enum makes the comment wrong, then how does that same usage not invalidate the "self explanatory" meaning of the enum? That tells me what we think of as "self explanatory" is a little bit fuzzy. If that makes sense. At the very least I think more comments at a high level ("these exported types mean this" "this exported function does this") and less comments about implementation details ("// fetch the job") will help out people down the line

Reply:
haha fair enough - dropping the comments here is a loose opinion weakly held. 

Happy to keep them

Reply:
If you ever run across the alphabetizing manifesto let me know plz